gdpr/impl_test.go

```diff
@@ -177,6 +177,80 @@ func TestMalformedConsent(t *testing.T) {
 }
 
 func TestAllowPersonalInfo(t *testing.T) {
+	tests := []struct {
+		description string
+		bidderName  openrtb_ext.BidderName
+		publisherID string
+		gdpr        Signal
+		consent     string
+		allowPI     bool
+	}{
+		{
+			description: "Allow PI - Non standard publisher",
+			bidderName:  openrtb_ext.BidderAppnexus,
+			publisherID: "appNexusAppID",
+			gdpr:        SignalYes,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     true,
+		},
+		{
+			description: "Allow PI - known vendor with No GDPR",
+			bidderName:  openrtb_ext.BidderPubmatic,
+			gdpr:        SignalNo,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     true,
+		},
+		{
+			description: "Allow PI - known vendor with Yes GDPR",
+			bidderName:  openrtb_ext.BidderPubmatic,
+			gdpr:        SignalYes,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     true,
+		},
+		{
+			description: "Don't allow PI - known vendor with Yes GDPR but vendor is without TCF1 purposes",
+			bidderName:  openrtb_ext.BidderAppnexus,
+			gdpr:        SignalYes,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     false,
+		},
+		{
+			description: "PI allowed according to host setting UserSyncIfAmbiguous - known vendor with ambiguous GDPR and empty consent",
+			bidderName:  openrtb_ext.BidderPubmatic,
+			gdpr:        SignalAmbiguous,
+			consent:     "",
+			allowPI:     true,
+		},
+		{
+			description: "Don't allow PI - known vendor with Yes GDPR and empty consent",
+			bidderName:  openrtb_ext.BidderPubmatic,
+			gdpr:        SignalYes,
+			consent:     "",
+			allowPI:     false,
+		},
+		{
+			description: "Don't allow PI - default vendor with Yes GDPR and empty consent",
+			bidderName:  openrtb_ext.BidderRubicon,
+			gdpr:        SignalYes,
+			consent:     "",
+			allowPI:     false,
+		},
+		{
+			description: "Don't allow PI - default vendor with ambiguous GDPR and non-empty consent",
+			bidderName:  openrtb_ext.BidderRubicon,
+			gdpr:        SignalAmbiguous,
+			consent:     "BOS2bx5OS2bx5ABABBAAABoAAAABBwAA",
+			allowPI:     false,
+		},
```

Comment:
I'm a bit confused by this case. What do I have wrong in my head?

The above test case should allow PI because we have an ambiguous signal with UsersyncIfAmbiguous set to true (which means do not enforce GDPR?).